src/backupNg/backupNg.spec.js

```diff
@@ -242,4 +232,64 @@ describe("backupNg", () => {
       expect(vmTask.data.id).toBe(vmIdWithoutDisks);
     });
   });
+
+  test("execute three times a rolling snapshot with 2 as retention & revert to an old state", async () => {
+    jest.setTimeout(7e4);
+    const vmId = await xo.createTempVm({
+      name_label: "XO Test Temp",
+      name_description: "Creating a temporary vm",
+      template: config.xoTestTemplateId,
+      VDIs: [
+        {
+          size: 1,
+          SR: config.srs.defaultSr,
+          type: "user",
+        },
+      ],
+    });
+
+    const scheduleTempId = randomId();
+    const { id: jobId } = await xo.createTempBackupNgJob({
+      ...defaultBackupNg,
+      vms: {
+        id: vmId,
+      },
+      schedules: {
+        [scheduleTempId]: defaultSchedule,
+      },
+      settings: {
+        ...defaultBackupNg.settings,
+        [scheduleTempId]: { snapshotRetention: 2 },
+      },
+    });
+
+    const schedule = await xo.getSchedule({ jobId });
+    expect(typeof schedule).toBe("object");
+    for (let i = 0; i < 3; i++) {
+      const oldSnapshots = xo.objects.all[vmId].snapshots;
+      await xo.call("backupNg.runJob", { id: jobId, schedule: schedule.id });
+      expect(xo.objects.all[vmId].snapshots).not.toEqual(oldSnapshots);
+    }
+
+    const { snapshots, videoram: oldVideoram } = xo.objects.all[vmId];
+    expect(snapshots.length).toBe(2);
+
+    const newVideoram = 16;
+    await xo.call("vm.set", { id: vmId, videoram: newVideoram });
+    await xo.waitObjectState(vmId, ({ videoram }) => {
+      expect(videoram).toBe(newVideoram.toString());
+    });
+
+    await xo.call("vm.revert", {
+      snapshot: xo.objects.all[vmId].snapshots[0],
+    });
+
+    await xo.waitObjectState(
+      vmId,
+      ({ current_operations: currentOperations }) => {
+        expect(Object.keys(currentOperations).length).toEqual(0);
```

Comment:
Is this test needed?

Reply:
This test is a way to wait the end of the reverting. Is this test useless ?

Reply:
I think I will remove this test. You are right. I can only wait the `videoram` updated.